src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -780,17 +783,47 @@ internal void Assign(IEnumerable<TopicPartitionOffset> partitions)
                 }
             }
 
-            ErrorCode err = Librdkafka.assign(handle, list);
+            ErrorCode err = ErrorCode.NoError;
+            Error error = null;
+
+            if (assignMethodErr != null) { err = assignMethodErr(handle, list); }
+            if (assignMethodError != null) { error = new Error(assignMethodError(handle, list)); }
+
             if (list != IntPtr.Zero)
             {
                 Librdkafka.topic_partition_list_destroy(list);
             }
-            if (err != ErrorCode.NoError)
+
+            if (err != ErrorCode.NoError) { throw new KafkaException(CreatePossiblyFatalError(err, null)); }
+            if (error != null && error.Code != ErrorCode.NoError) { throw new KafkaException(error); }
+        }
+
+        internal void Assign(IEnumerable<TopicPartitionOffset> partitions)
+            => AssignImpl(partitions, Librdkafka.assign, null);
+
+        internal void IncrementalAssign(IEnumerable<TopicPartitionOffset> partitions)
+            => AssignImpl(partitions, null, Librdkafka.incremental_assign);
+
+        internal void IncrementalUnassign(IEnumerable<TopicPartitionOffset> partitions)
+            => AssignImpl(partitions, null, Librdkafka.incremental_unassign);
+
+        internal bool AssignmentLost
+        {
+            get
             {
-                throw new KafkaException(CreatePossiblyFatalError(err, null));
+                ThrowIfHandleClosed();
+                return (Librdkafka.assignment_lost(handle) != IntPtr.Zero);
             }
         }
 
+        internal string RebalanceProtocol
+        {
+            get
+            {
+                ThrowIfHandleClosed();
+                return Util.Marshal.PtrToStringUTF8(Librdkafka.rebalance_protocol(handle));
```

Comment:
It will return NULL if used after calling Close()

Reply:
changed to return null in that case.